File: transaction_validator/src/main/java/com/example/transaction_validator/entities/Envelope.java
package com.example.transaction_validator.entities;

import java.time.LocalDateTime;
import java.util.List;

import com.example.transaction_validator.entities.enums.EnvelopeStatus;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor

public class Envelope {
    @Id
    @Column(name = "transactionId")
    private Long envelopeId;

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    @JoinColumn(name = "transactionId")
    private Transaction transaction;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private EnvelopeStatus status = EnvelopeStatus.PENDING; // "COMPLETED" or "PENDING"

    private LocalDateTime signedAt;

    @OneToMany(mappedBy = "envelope", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<Signer> signers;
}

File: frontend/src/app/models/Envelope.ts
export type Envelope = {
    envelopeId: number;
    transactionId: number;
    status: 'COMPLETED' | 'PENDING' | string;
    signedAt: string;
    signers: Signer[];
}

export type Signer = {
    signerId: number;
    envelopeId: number;
    envelopeStatus: string;
    signerEmail: string;
    signedAt: string | null;
    userId: number;
}

export type Transaction = {
    transactionID: number;
    amount: number;
    timestamp: string | null;
    senderAccountId: number;
    receiverAccountId: number;
    status: "INCOMPLETE" | "APPROVED" | "FLAGGED" | string;
}

